src/ape/managers/query.py

```diff
@@ -53,6 +22,48 @@ class QueryManager:
         chain.blocks.query()
     """
 
-    def query(self, query: "Query") -> pd.DataFrame:
-        """ """
-        pass
+    plugin_manager: PluginManager
+    """A reference to the global plugin manager."""
+
+    @cached_property
+    def query_engines(self) -> List[QueryAPI]:
+        return [engine for _, engine in self.plugin_manager.query_engines]
+
+    def __getitem__(self, engine_name: str) -> QueryAPI:
+        """
+        Get an engine by name.
+
+        Args:
+            engine_name (str): The name of the engine to get.
+
+        Returns:
+            :class:`~ape.api.query.QueryAPI`
+        """
+        if engine_name not in self.plugin_manager.query_engines:
+            raise QueryEngineException(engine_name)
+
+        return self.plugin_manager.query_engines[engine_name]
+
+    def query(self, query: Query) -> pd.DataFrame:
+        """
+        Args:
+            query (``Query``): The type of query to execute
+
+        Raises: :class:`~ape.exceptions.QueryEngineException`: When the
+            query.engine_to_use is invalid or inaccessible
+
+        Returns:
+            pandas.DataFrame
+        """
+
+        # Get heuristics from all the query engines to perform this query
+        # NOTE: `self.query_engines` should be `cached_property` of loading
+        #       `[engine for _, engine in PluginManager.query_engines]`
+        estimates = map(lambda qe: (qe, qe.estimate_query(query)), self.query_engines)
```

Comment:
If changed to a dict
```suggestion
        estimates = map(lambda qe: (qe, qe.estimate_query(query)), self.query_engines.values())
```